cmake/kokkos_test_cxx_std.cmake

```diff
@@ -86,6 +86,17 @@ ELSE()
   MESSAGE(FATAL_ERROR "Unknown C++ standard ${KOKKOS_CXX_STANDARD} - must be 14, 17, or 20")
 ENDIF()
 
+# Enforce that we can compile a simple C++14 program
+
+TRY_COMPILE(CAN_COMPILE_CPP14
+    ${KOKKOS_TOP_BUILD_DIR}/corner_cases
+    ${KOKKOS_SOURCE_DIR}/cmake/compile_tests/cplusplus14.cpp
+)
+if (NOT CAN_COMPILE_CPP14)
+    UNSET(CAN_COMPILE_CPP14 CACHE) #make sure CMake always re-runs this
+    MESSAGE(FATAL_ERROR "Kokkos requires C++14 or newer!")
```

Comment:
The error message should minimally detail the fact that CMake is detecting C++14 as valid, but a compiler sanity check is failing - and possibly recommend some things to look at.

Reply:
Tried to make the error less cryptic by suggesting the compiler may not be set up correctly.